notebooks/peerscout/peerscout-get-editor-pubmed-paper-ids.ipynb

```diff
@@ -77,102 +75,188 @@
+   ],
+   "outputs": [],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "source": [
+    "merged_editor_parsed_pubmed_links_df = editor_parsed_pubmed_links_df.merge(\n",
+    "    existing_editor_ids_and_pubmed_url_df,\n",
+    "    how='left',\n",
+    "    on='person_id',\n",
+    "    suffixes=('', '_existing')\n",
+    ")\n",
+    "\n",
+    "print(\"merged_editor_parsed_pubmed_links_df length: \", len(merged_editor_parsed_pubmed_links_df))\n",
+    "merged_editor_parsed_pubmed_links_df.head(3)"
+   ],
    "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
    "source": [
-    "remaining_editor_parsed_pubmed_links_df = editor_parsed_pubmed_links_df[\n",
+    "editors_with_changed_pubmed_url_df = merged_editor_parsed_pubmed_links_df[\n",
+    "    (merged_editor_parsed_pubmed_links_df['pubmed_url_existing'].notnull())\n",
+    "    &\n",
+    "    (\n",
+    "        merged_editor_parsed_pubmed_links_df['pubmed_url']\n",
+    "        !=\n",
+    "        merged_editor_parsed_pubmed_links_df['pubmed_url_existing']\n",
+    "    )\n",
+    "].drop(columns=['pubmed_url_existing', 'imported_timestamp'])\n",
+    "\n",
+    "print(\"editors_with_changed_pubmed_url_df length: \", len(editors_with_changed_pubmed_url_df))\n",
+    "editors_with_changed_pubmed_url_df.head(3)"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "editors_with_not_currently_updated_info = pd.DataFrame()\n",
+    "temp_df = pd.DataFrame()\n",
+    "for index, row in merged_editor_parsed_pubmed_links_df.iterrows():\n",
+    "    day_diff = abs(\n",
+    "        datetime.date(row['imported_timestamp'])-datetime.date(datetime.today())\n",
+    "    ).days\n",
+    "    if day_diff > 15:\n",
+    "        temp_df = row\n",
+    "    editors_with_not_currently_updated_info = editors_with_not_currently_updated_info.append(temp_df)\n",
+    "\n",
+    "print(\"editors_with_not_currently_updated_info length: \", len(editors_with_not_currently_updated_info))\n",
+    "editors_with_not_currently_updated_info.head(3)"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "new_added_editors_df = editor_parsed_pubmed_links_df[\n",
     "    ~editor_parsed_pubmed_links_df['person_id'].isin(existing_editor_ids_set)\n",
     "]\n",
-    "print(len(remaining_editor_parsed_pubmed_links_df))\n",
-    "remaining_editor_parsed_pubmed_links_df.head(3)"
-   ]
+    "\n",
+    "print(\"new_added_editors_df length: \", len(new_added_editors_df))\n",
+    "new_added_editors_df.head(3)"
+   ],
+   "outputs": [],
+   "metadata": {}
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "source": [
+    "remaining_editor_parsed_pubmed_links_df = pd.concat([\n",
+    "    new_added_editors_df,\n",
```

Comment:
One disadvantage of concatenating dataframes from different conditions is that you would want to make sure that there is no overlap. Otherwise the same person id could be included multiple times. That is probably the case for the first two, but not sure about `editors_with_not_currently_updated_info`? (e.g. if it had changed now, but also hasn't been updated for a while)

If you instead just select the rows, either using a single more complex condition, or chained, then this might be easier.

Perhaps an alternative, if you don't like the dataframe selection, could also be to have a final selection based on a set of person ids, similar to before. Something like:

```python
selected_person_ids = (
    set(new_added_editors_df['person_id'))
    | set(editors_with_changed_pubmed_url_df['person_id'))
    | set(editors_with_not_currently_updated_info['person_id'))
)
remaining_editor_parsed_pubmed_links_df = editor_parsed_pubmed_links_df[
    editor_parsed_pubmed_links_df['person_id'].isin(selected_person_ids)
]
```